src/Analytics/AnomalyDetection/Detectors/CredentialStuffingDetector.php

```
@@ -0,0 +1,290 @@
+        ) );
+        $globalAttempts[] = [ 'ts' => $nowTs, 'ip' => $ip, 'success' => (bool) $success ];
+
+        Cache::put( $globalKey, $globalAttempts, now()->addMinutes( $window ) );
+    }
+
+    /**
+     * Check for credential stuffing from specific IP.
+     *
+     * @param  array<string, mixed>  $data
+     */
+    protected function checkCredentialStuffing( string $ip, array $data ): ?Anomaly
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$data'. (undefined)

(UnusedFormalParameter)